src/makeCalculator.js

```diff
@@ -37,7 +37,64 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate(callback, operand) {
+      switch (callback) {
+        case this.add:
+          this.add(operand);
+
+          break;
+
+        case this.subtract:
+          this.subtract(operand);
+
+          break;
+
+        case this.multiply:
+          this.multiply(operand);
+
+          break;
+
+        case this.divide:
+          this.divide(operand);
+
+          break;
+
+        default:
+          break;
+      }
+
+      return this;
+    },
```

Comment:
there is unnecessary code. if you will have two parameters in each method and use callback in your operade method - the code becomes short.